internal/querynodev2/segments/segment.go

```diff
@@ -1156,26 +1156,28 @@ func (s *LocalSegment) WarmupChunkCache(ctx context.Context, fieldID int64, mmap
 			return nil, nil
 		}).Await()
 	case "async":
-		GetWarmupPool().Submit(func() (any, error) {
-			// bad implemtation, warmup is async at another goroutine and hold the rlock.
-			// the state transition of segment in segment loader will blocked.
-			// add a waiter to avoid it.
-			s.ptrLock.BlockUntilDataLoadedOrReleased()
-			if !s.ptrLock.RLockIf(state.IsNotReleased) {
+		go func() {
```

Comment:
why do we add a go routine outside a pool of thread?

Reply:
there will be too many go routine if pool is full

Reply:
let's move this get warm pool from go to cpp.

We can use a another folly pool instead of async here.

folly support unbounded queue tag


    // Create an unbounded task queue
    auto taskQueue = std::make_shared<folly::LifoSemMPMCQueue<folly::CPUThreadPoolExecutor::CPUTask>>(
        folly::UnboundedQueueTag());

    // Create a CPUThreadPoolExecutor with bounded threads and an unbounded task queue
    folly::CPUThreadPoolExecutor executor(numThreads, std::move(taskQueue));

Reply:
And we should use only
cpunumber / 2 for warmup to avoid extra high cpu usage